### PR TITLE
[Mobile] - Remove old Layout Picker

### LINKED ISSUE #2936: Remove old Page Layout Picker
After the release of the new native page layout picker that's built-in in the WordPress apps, the previous picker in Gutenberg is no longer used and should be removed.

Old picker | New picker
-|-
<kbd></kbd> | <kbd></kbd>

### PR DESCRIPTION
**Pull request by geriux**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2936

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2921
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/15550
`WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/13665

## Description
Removes unused code of the page layout picker. A native layout picker was implemented in the WordPress apps so this is no longer being used.

Removes:
* __experimentalPageTemplatePicker
* PageTemplatePicker
* Default layouts
* Layout picker Tooltip and its persistence logic
* ModalHeaderBar component
* Moves the Preview component to its own (mobile only)

It also fixes some tests issues related to the recently introduced `alignmentHelpers`

## How has this been tested?
On the WordPress app (builds available in each PR)

### Case 1
- Go to My site tab
- Tap on the Create new button (bottom right)
- Select Site page
- **Expect** to see: 
<kbd></kbd>
- Tap on any layout
- Tap on the Preview button
- **Expect** to see the preview of the layout
- Tap on Create page
- **Expect** to see the editor opened with the selected layout

### Case 2
- Go to Pages
- Tap on the Create button (bottom right)
- **Expect** to see: 
<kbd></kbd>
- Tap on create Blank page
- **Expect** to see the editor without any content

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by antonis — APPROVED** [review]
Great work @geriux 👍
I tested the app on a Pixel 2XL with Android 11 and behaves as expected.
The code changes are consistent and as described remove:
* The whole old Page Template Picker (`page-template-picker` folder) functionality except `BlockPreview` that has been moved
* The `ModalHeaderBar` (`modal-header-bar` folder)
* The [Modal Layout Picker flag for Android](https://github.com/WordPress/gutenberg/pull/24061/files)
* The [Starter Page Template Tooltip code](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2216)

**Review by chipsnyder — COMMENTED** [review]
Thanks for taking this on @geriux!

There's an enum case for `modalLayoutPicker` under capabilities that I think we can remove now too. Could we include that deletion in this PR as well? 

https://github.com/WordPress/gutenberg/blob/db9da4a5d2a0ad49915c96dfa2c0778bd6c67b59/packages/react-native-bridge/ios/GutenbergBridgeDelegate.swift#L24

File: packages/edit-post/src/components/layout/index.native.js [hunk 1]
```diff
@@ -161,11 +161,9 @@ export default compose( [
 			'core/editor'
 		);
 		const { getEditorMode } = select( editPostStore );
-		const { getSettings } = select( 'core/block-editor' );
 		return {
 			isReady: isEditorReady(),
 			mode: getEditorMode(),
-			modalLayoutPicker: getSettings( 'capabilities' ).modalLayoutPicker,
```
Comment (chipsnyder) [inline packages/edit-post/src/components/layout/index.native.js:183]: Good catch on these too!

**Review by chipsnyder — APPROVED** [review]
Thanks for taking care of this! Tested with https://github.com/wordpress-mobile/WordPress-iOS/pull/15550#pullrequestreview-563750901

**Pull request merged**